cypress/integration/collections_spec.js

```diff
@@ -62,40 +62,42 @@ describe('Dashboard Collections Page', () => {
     it('should add a new collection', () => {
       const name = 'TESTCOLLECTION';
       const version = '006';
-      const duplicateHandling = 'replace';
+      // Test collection loaded by cy.fixture
+      let collection;
 
-      cy.visit('/#/collections');
+      cy.server();
+      cy.route('POST', '/collections').as('postCollection');
 
+      // On the Collections page, click the Add Collection button
+      cy.visit('/#/collections');
       cy.contains('.heading--large', 'Collection Overview');
-      cy.contains('a', 'Add a Collection').as('addCollection');
-      cy.get('@addCollection').should('have.attr', 'href', '#/collections/add');
-      cy.get('@addCollection').click();
-
-      // fill the form and submit
-      cy.fixture('TESTCOLLECTION___006.json').then((collection) => {
-        cy.editJsonTextarea({ data: collection });
-      });
+      cy.contains('a', 'Add a Collection').click();
+
+      // Fill the form with the test collection JSON and submit it
+      cy.hash().should('eq', '#/collections/add');
+      cy.fixture('TESTCOLLECTION___006.json')
+        .then((json) => {
+          cy.editJsonTextarea({ data: json });
+          // Capture the test collection so we can confirm below that it was
+          // properly persisted after form submission.
+          collection = json;
+        });
       cy.contains('form button', 'Submit').click();
 
-      // displays the new collection
+      // After POSTing the new collection, make sure we GET it back
+      cy.wait('@postCollection')
+        .then((xhr) =>
+          cy.request({
+            method: 'GET',
+            url: `${new URL(xhr.url).origin}/collections/${name}/${version}`,
+            headers: xhr.request.headers
+          }))
+        .then((response) => expect(response.body).to.deep.equal(collection));
+
+      // Display the new collection
+      cy.hash().should('eq', `#/collections/collection/${name}/${version}`);
       cy.contains('.heading--xlarge', 'Collections');
       cy.contains('.heading--large', `${name} / ${version}`);
-      cy.url().should('include', `#/collections/collection/${name}/${version}`);
-
-      // verify the collection's properties by looking at the Edit page
-      cy.contains('a', 'Edit').click();
-      cy.get('form .ace_content')
-        .within(() => {
-          cy.contains(`"name": "${name}"`);
-          cy.contains(`"version": "${version}"`);
-          cy.contains(`"dataType": "${name}"`);
-          cy.contains(`"duplicateHandling": "${duplicateHandling}"`);
-        });
-
-      // verify the new collection is added to the collections list
-      cy.contains('a', 'Back to Collections').click();
```

Comment:
I don't think we want to lose this test assertion proving that the collection shows up in the collections list

Reply:
I'm on the fence with that suggestion because that's arguably not the responsibility of this particular test, which is simply to make sure that when the user adds a new the collection it is persisted and the user is taken to the page for the new collection itself.

Navigating back to the collections list page and then testing that the collections list page includes the newly added collection (or more generally, the page lists all persisted collections) is arguably the responsibility of a different test, and the test "should display expected MMT Links for collections list" does indeed test that case by expecting a list of all known collections (in the context of the test) to appear.

Given that the test here proves the collection is persisted, and the "MMT Links" test proves that all persisted collections are listed, I find the lines I've removed here are unnecessary.

However, if you feel strongly about this, I'm not adamantly opposed to it.

Reply:
Hmm, in theory the tests are slightly different I think? The test in "MMT links" just proves that the correct collections show up for what already is persisted in the API. This test is proving that a collection is not only added correctly so that it is returned by a direct /GET request, but persisted in such a way that it shows up in **listing** /GET requests. This seems critical given that the actual persistence layer behind the listing requests (Elasticsearch) is different than behind a single GET request (Dynamo). I know that our dashboard code tests should be ignorant of that distinction, but it feels like they should test all of the expected UI changes associated with an operation such as "add a collection", which would have a nice side effect of also testing that the persistence to both layers is working properly. So I would vote in favor of including this test

Reply:
Ah, I wasn't aware of that distinction, so I agree that this makes sense, and I will update per your request.

Reply:
@markdboyd I restored the extra check, along with leveraging CMR response mocking, which has resolved the flakiness of the "add collection" test.